x/nodes/keeper/reward.go

```diff
@@ -17,7 +18,32 @@ func (k Keeper) RewardForRelays(ctx sdk.Ctx, relays sdk.BigInt, address sdk.Addr
 			return sdk.ZeroInt()
 		}
 	}
-	coins := k.RelaysToTokensMultiplier(ctx).Mul(relays)
+
+	var coins sdk.BigInt
+
+	//check if PIP22 is enabled, if so scale the rewards
+	if k.Cdc.IsAfterNamedFeatureActivationHeight(ctx.BlockHeight(), codec.RSCALKey) {
+		//grab stake
+		validator, found := k.GetValidator(ctx, address)
+		if !found {
+			ctx.Logger().Error(fmt.Errorf("no validator found for address %s; at height %d\n", address.String(), ctx.BlockHeight()).Error())
+			return sdk.ZeroInt()
+		}
+
+		stake := validator.GetTokens()
+		//floorstake to the lowest bin multiple or take ceiling, whicherver is smaller
+		flooredStake := sdk.MinInt(stake.Sub(stake.Mod(k.ServicerStakeFloorMultiplier(ctx))), (k.ServicerStakeWeightCeiling(ctx).Sub(k.ServicerStakeWeightCeiling(ctx).Mod(k.ServicerStakeFloorMultiplier(ctx)))))
+		//Convert from tokens to a BIN number
+		Bin := flooredStake.Quo(k.ServicerStakeFloorMultiplier(ctx))
+		//calculate the weight value, weight will be a floatng point number so cast to DEC here and then truncate back to big int
+		weight := Bin.ToDec().FracPow(k.ServicerStakeFloorMultiplierExponent(ctx), 100).Quo(k.ServicerStakeWeightMultiplier(ctx))
```

Comment:
Should the hardcoded `100` argument here be a parameter as well?

Reply:
If not, it should still be a constant declared globally with enough documentation to be able to go back to why the value was hard coded in the first place.

Reply:
Yes this should be configurable, so declared globally. It is used to avoid overflows when finding the nth root of a decimal number. having 100 ensures that the denominator of the decimal number can never be smaller than 100th's. 

Were would be best to declare this?, personally don't think it needs global scope as it is local to both reward and slashing, so only needs to be available to nodes keeper.

Reply:
addressed in [7f6fbed](https://github.com/pokt-network/pocket-core/pull/1449/commits/7f6fbed326cae72229041a71f1735055a82f425f)